arroyo-worker/src/connectors/polling_http.rs

```diff
@@ -270,17 +277,8 @@ where
                                 let iter = self.deserializer.deserialize_slice(&buf).await;
 
                                 for record in iter {
-                                    match record {
-                                        Ok(value) => {
-                                            ctx.collect(Record {
-                                                timestamp: SystemTime::now(),
-                                                key: None,
-                                                value,
-                                            }).await;
-                                        }
-                                        Err(e) => {
-                                            ctx.report_user_error(e).await;
-                                        }
+                                    if let Err(e) = collect_source_record(ctx, SystemTime::now(), record, &self.bad_data, &mut self.rate_limiter).await {
```

Comment:
I think this needs to be updated to remove the existing bad data handling code

Reply:
It looks like most sources panic on user errors, except for this one and SSE. Is there a reason for that or should we make them all consistent?

Reply:
I've removed this and made all the sources consistent in handling user errors.